### PR TITLE
[docs] EAS Build with submodules

### PR DESCRIPTION
**Pull request by wkozyra95**
# Why

https://linear.app/expo/issue/ENG-1528/documentation-for-how-to-use-git-submodules

# How

# Test Plan

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/build-reference/how-tos.md [hunk 1]
```diff
@@ -83,3 +83,28 @@ e.g.
   }
 }
 ```
+
+## How to use git submodules
+
+Create a [secret](/build-reference/variables/#using-secrets-in-environment-variables) with base64 encoded private ssh key that have permission to access submodules repositories and add `eas-build-pre-install` npm hook that is checking out those submodules e.g.
```
Comment (dsokal) [inline docs/pages/build-reference/how-tos.md:89]:
```suggestion
Create a [secret](/build-reference/variables/#using-secrets-in-environment-variables) with a base64 encoded private ssh key that has permission to access submodules repositories and add the `eas-build-pre-install` npm hook that is checking out those submodules, e.g.
```

File: docs/pages/build-reference/how-tos.md [hunk 2]
```diff
@@ -83,3 +83,28 @@ e.g.
...
+
+```bash
+#!/usr/bin/env bash
+
+mkdir -p ~/.ssh
+
+# uncomment line bellow and specify url for remote "origin" if your submodules
```
Comment (dsokal) [inline docs/pages/build-reference/how-tos.md:96]:
```suggestion
# uncomment the line below and replace the origin's URL if your submodules
```

**Review by dsokal — APPROVED** [review]

File: docs/pages/build-reference/how-tos.md [hunk 3]
```diff
@@ -83,3 +83,28 @@ e.g.
   }
 }
 ```
+
+## How to use git submodules
+
+Create a [secret](/build-reference/variables/#using-secrets-in-environment-variables) with a base64 encoded private ssh key that has permission to access submodules repositories and add the `eas-build-pre-install` npm hook that is checking out those submodules, e.g.
```
Comment (brentvatne) [inline docs/pages/build-reference/how-tos.md:89]:
```suggestion
First, create a [secret](/build-reference/variables/#using-secrets-in-environment-variables) with a base64 encoded private SSH key that has permission to access submodules repositories. Next, add an `eas-build-pre-install` npm hook to check out those submodules, for example:
```

File: docs/pages/build-reference/how-tos.md [hunk 4]
```diff
@@ -83,3 +83,28 @@ e.g.
...
+
+```bash
+#!/usr/bin/env bash
+
+mkdir -p ~/.ssh
+
+# uncomment the line below and replace the origin's URL if your submodules
+# are defined with relative urls in .gitmodules
```
Comment (brentvatne) [inline docs/pages/build-reference/how-tos.md:97]: this could use more context

**Review by brentvatne — APPROVED** [review]

**Pull request merged**